source/slang/ir.h

```diff
@@ -114,10 +115,24 @@ enum IRDecorationOp : uint16_t
     kIRDecorationOp_TargetIntrinsic,
 };
 
+// represents an object allocated in an IR memory pool 
+struct IRObject
+{
+    bool isDestroyed = false;
+    virtual void dispose()
+    {
+        isDestroyed = true;
+    }
+    virtual ~IRObject()
```

Comment:
Should `dispose` be asserting that `isDestroyed == false`? Should the destructor be asserting that `isDestroyed == true`?